aspnetcore/security/authentication/cookie.md

```diff
@@ -14,173 +14,299 @@ uid: security/authentication/cookie
-       {
-           IsPersistent = true
-       });
-   ```
+    "MyCookieMiddlewareInstance",
+    principal,
+    new AuthenticationProperties
+    {
+        IsPersistent = true
+    });
+```
 
-This code snippet will create an identity and corresponding cookie which will survive through browser closures. Any sliding expiration settings previously configured via [cookie options](xref:security/authentication/cookie#security-authentication-cookie-options) will still be honored, if the cookie expires whilst the browser is closed the browser will clear it once it is restarted.
+The `AuthenticationProperties` class, used in the preceding code snippet, resides in the `Microsoft.AspNetCore.Http.Authentication` namespace.
 
-<a name=security-authentication-absolute-expiry></a>
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-<!-- literal_block {"ids": ["security-authentication-absolute-expiry"], "linenos": false, "names": ["security-authentication-absolute-expiry"], "xml:space": "preserve", "language": "csharp"} -->
+```csharp
+await HttpContext.SignInAsync(
+    "MyCookieMiddlewareInstance",
+    principal,
+    new AuthenticationProperties
+    {
+        IsPersistent = true
+    });
+```
+
+The `AuthenticationProperties` class, used in the preceding code snippet, resides in the `Microsoft.AspNetCore.Authentication` namespace.
+
+---
+
+This code snippet creates an identity and corresponding cookie which survives through browser closures. Any sliding expiration settings previously configured via [cookie options](xref:security/authentication/cookie#security-authentication-cookie-options) are still honored. If the cookie expires whilst the browser is closed, the browser clears it once it is restarted.
+
+<a name="security-authentication-absolute-expiry"></a>
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 ```csharp
 await HttpContext.Authentication.SignInAsync(
-       "MyCookieMiddlewareInstance",
-       principal,
-       new AuthenticationProperties
-       {
-           ExpiresUtc = DateTime.UtcNow.AddMinutes(20)
-       });
-   ```
+    "MyCookieMiddlewareInstance",
+    principal,
+    new AuthenticationProperties
+    {
+        ExpiresUtc = DateTime.UtcNow.AddMinutes(20)
+    });
+```
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+```csharp
+await HttpContext.SignInAsync(
+    "MyCookieMiddlewareInstance",
+    principal,
+    new AuthenticationProperties
+    {
+        ExpiresUtc = DateTime.UtcNow.AddMinutes(20)
+    });
+```
+
+---
 
-This code snippet will create an identity and corresponding cookie which will last for 20 minutes. This ignores any sliding expiration settings previously configured via [cookie options](xref:security/authentication/cookie#security-authentication-cookie-options).
+This code snippet creates an identity and corresponding cookie which lasts for 20 minutes. This ignores any sliding expiration settings previously configured via [cookie options](xref:security/authentication/cookie#security-authentication-cookie-options).
```

Comment:
Yeah if you want to turn off slidingRefresh, they need to do it explicitly via options.SlidingExpiration = false.  IsPersistent comes from AuthenticationProperties and is something the callee controls.